core/domain/user_services.py

```diff
@@ -1414,10 +1439,23 @@ def update_email_preferences(
     email_preferences_model.subscription_notifications = (
         can_receive_subscription_email)
     email_preferences_model.site_updates = can_receive_email_updates
-    platform_bulk_email_services.add_or_update_user_status(
-        get_email_from_user_id(user_id), can_receive_email_updates)
+    email = get_email_from_user_id(user_id)
+    if update_bulk_email_db:
+        try:
+            bulk_email_services.add_or_update_user_status(
+                email, can_receive_email_updates)
+        except Exception as error_message:
+            error_message = ast.literal_eval(
+                python_utils.UNICODE(error_message))
+            if error_message['title'] == 'Forgotten Email Not Subscribed':
```

Comment:
Perhaps link to a doc that explains the "Forgotten Email Not Subscribed" string, or explain where it comes from exactly. 

If it's specific to the bulk email service then perhaps have bulk_email_services.add_or_update_user_status() raise a different type of exception instead that gets caught here in a provider-agnostic way. The phrase that's currently here seems pretty specific to a particular provider, but this layer of the codebase should be generic. You can also name the custom error class such that it explains the context of the error (which currently isn't really clear).

Reply:
Done, moved this exception catch to inside the mailchimp service file.